File: packages/java/src/main/java/com/ummalqura/HijriDate.java
package com.ummalqura;

/**
 * Hijri (Umm al-Qura) date representation
 */
public class HijriDate {
    private final int year;
    private final int month;
    private final int day;

    public HijriDate(int year, int month, int day) {
        this.year = year;
        this.month = month;
        this.day = day;
    }

    public int getYear() { return year; }
    public int getMonth() { return month; }
    public int getDay() { return day; }
    public String getCalendar() { return "hijri-ummalqura"; }

    @Override
    public String toString() {
        return String.format("%d-%02d-%02d", year, month, day);
    }
}

File: packages/php/src/HijriDate.php
<?php

namespace UmmAlQura;

class HijriDate
{
    public function __construct(
        private int $year,
        private int $month,
        private int $day
    ) {}

    public function getYear(): int { return $this->year; }
    public function getMonth(): int { return $this->month; }
    public function getDay(): int { return $this->day; }
    public function getCalendar(): string { return 'hijri-ummalqura'; }

    public function toArray(): array
    {
        return [
            'year' => $this->year,
            'month' => $this->month,
            'day' => $this->day,
            'calendar' => $this->getCalendar(),
        ];
    }
}
